src/riak_kv_pb_bucket.erl

```diff
@@ -96,28 +94,60 @@ encode(Message) ->
     {ok, riak_pb_codec:encode(Message)}.
 
 %% @doc process/2 callback. Handles an incoming request message.
-process(#rpblistbucketsreq{type = Type, timeout=T, stream=S}=Req, State) ->
-    case check_bucket_type(Type) of
-        {ok, GoodType} ->
-            do_list_buckets(GoodType, T, S, Req, State);
-        error ->
-            {error, {format, "No bucket-type named '~s'", [Type]}, State}
+process(#rpblistbucketsreq{type = Type, timeout = T, stream = S} = Req, State) ->
+    Class = case S of
+        true ->
+            {riak_kv, stream_list_buckets};
+        _ ->
+            {riak_kv, list_buckets}
+    end,
+    Accept = riak_core_util:job_class_enabled(Class),
```

Comment:
Yes and no - there's certainly _some_ commonality, but the `process` (for PB) and `forbidden` (for WM) functions differ enough for various operations that it could get fairly convoluted.  It certainly could be changed to a different pattern, but I wanted to keep complexity down for the first pass.
I also opted to try to keep more of the logic out at the edge for clarity, assuming there'll be a good deal of refactoring for the 2.3 jobs API.